addons/stock_calendar/models/procurement.py

```diff
@@ -97,31 +91,29 @@ def _prepare_orderpoint_procurement(self, cr, uid, orderpoint, product_qty, date
             'group_id': group or orderpoint.group_id.id,
             }
 
-    def _get_previous_dates(self, cr, uid, orderpoint, start_date = False, context=None):
+    def _get_previous_dates(self, orderpoint, start_date=False):
         """
         Date should be given in utc
         """
-        calendar_obj = self.pool.get('resource.calendar')
-        att_obj = self.pool.get('resource.calendar.attendance')
-        context = context or {}
-        context['no_round_hours'] = True
+        Resourcecalendar = self.env['resource.calendar'].with_context(no_round_hours=True)
+        ResourceCalendarAttendance = self.env['resource.calendar.attendance'].with_context(no_round_hours=True)
         # Date should be converted to the correct timezone
-        start_date = self._convert_to_tz(cr, uid, start_date, context=context)
+        start_date = self.with_context(no_round_hours=True)._convert_to_tz(start_date)
```

Comment:
can you check fields.Datetime.context_timestamp instead of this _convert_to_tz. do same thing..